R/produce_source_extract_tests.R

```diff
@@ -28,13 +29,22 @@ produce_source_extract_tests <- function(data,
                                          max_min_vars = c(
                                            "record_keydate1", "record_keydate2",
                                            "cost_total_net", "yearstay"
-                                         )) {
+                                         ),
+                                         hscp_var = TRUE) {
   test_flags <- data %>%
     # use functions to create HB and partnership flags
     create_demog_test_flags() %>%
     create_hb_test_flags(.data$hbtreatcode) %>%
-    create_hb_cost_test_flags(.data$hbtreatcode, .data$cost_total_net) %>%
-    create_hscp_test_flags(.data$hscp) %>%
+    create_hb_cost_test_flags(.data$hbtreatcode, .data$cost_total_net)
+
+  if (!hscp_var) {
+    test_flags <- test_flags
+  } else {
+    test_flags %>%
+      create_hscp_test_flags(.data$hscp)
+  }
```

Comment:
We don't really need an `else` - it's just if `hscp_var` is TRUE add the hscp flags, otherwise do nothing.
```suggestion
  if (hscp_var) {
    test_flags <- create_hscp_test_flags(test_flags, .data$hscp)
  }
```